numpy/core/src/umath/dispatching.c

```diff
@@ -742,14 +742,15 @@ promote_and_get_info_and_ufuncimpl(PyUFuncObject *ufunc,
     }
     info = promote_and_get_info_and_ufuncimpl(ufunc,
             ops, signature, new_op_dtypes, NPY_FALSE);
+    for (int i = 0; i < ufunc->nargs; i++) {
+        Py_XDECREF(new_op_dtypes[i]);
+    }
```

Comment:
The move is harmless. This was super painful to track down, but fortunately an easy fix:  The `[i]` was missing, so the `DECREF` was working on the stack and nothing happened.
(It is basically luck that I found this one, since the total ref-count doesn't change here, so `pytest-leaks` doesn't really find it!)